File: Python/July_problems/problem1518_7th_july_2024.py
def numWaterBottles(numBottles: int, numExchange: int) -> int:
    result = 0
    empty_bottel = 0
    while numBottles > 0:
        result += numBottles
        empty_bottel += numBottles
        numBottles = empty_bottel//numExchange
        empty_bottel -= numExchange*numBottles
    return result


def problem1518():
    pass


if __name__ == "__main__":
    problem1518()

File: CPP/July_Problems/problem1518_7th_july_2024.cpp
#include <iostream>

using namespace std;

int numWaterBottles(int numBottles, int numExchange)
{
    int result = 0, empty_bottel = 0;
    while (numBottles > 0)
    {
        result += numBottles;
        empty_bottel += numBottles;
        numBottles = (int)(empty_bottel / numExchange);
        empty_bottel = empty_bottel - (numBottles * numExchange);
    }
    return result;
}

int main()
{
    cout << numWaterBottles(9, 3);
    return 0;
}